entrypoint.d/45-geoip

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import re
+import sys
+
+from doodbalib import logger
+
+from plumbum import BG, local
+from plumbum.cmd import geoipupdate
+
+GEOIP_ACCOUNT_ID = local.env["GEOIP_ACCOUNT_ID"]
+GEOIP_LICENSE_KEY = local.env["GEOIP_LICENSE_KEY"]
+
+if GEOIP_ACCOUNT_ID not in {"", "0"} and GEOIP_LICENSE_KEY not in {"", "0"}:
+    # see https://dev.maxmind.com/geoip/geoipupdate/
+    logger.info("Activating GeoIP/GeoLite2 updates")
+    logger.info(
+        "This product includes GeoLite2 data created by MaxMind, "
+        "available from https://www.maxmind.com"
+    )
+    # Obtain current GeoIP configuration
+    geoip_file = local.path("/opt/odoo/auto/geoip/GeoIP.conf")
+    geoip_conf = geoip_file.read()
+    # Patch it with provided account and license keys
+    geoip_conf = re.sub(
+        r"(^|\n)AccountID\s*[^\n]*(\n|$)",
+        "\\1AccountID {}\\2".format(GEOIP_ACCOUNT_ID),
+        geoip_conf,
+    )
+    geoip_conf = re.sub(
+        r"(^|\n)LicenseKey\s*[^\n]*(\n|$)",
+        "\\1LicenseKey {}\\2".format(GEOIP_LICENSE_KEY),
+        geoip_conf,
+    )
```

Comment:
Before we had this:

```suggestion
    geoip_conf = re.sub(r"^AccountID\s*.*$", "AccountID {}".format(GEOIP_ACCOUNT_ID))
    geoip_conf = re.sub(r"^LicenseKey\s*.*$", "LicenseKey {}".format(GEOIP_LICENSE_KEY))
```

It seems more readable. Wasn't it enough? Keep in mind that AFAIK `^` and `$` match newlines if the regexp is not in multiline mode.

Reply:
as far as i remember this did not work:
* re.sub needs a third argument (the source string)
* your regex didn't match LicenseKey and AccoundID in all places it could appear (regex ^ applied on string only matches the beginning of the string (not beginning of a line), the same for `$`(end of string not newline) )